### PR TITLE
feat(analytics): Add an amplitude logging hook

### PR DESCRIPTION
**Pull request by adhiraj**
This is mostly a refactor, just adds a hook for logging things to amplitude.

### CODE REVIEW

File: src/sentry/static/sentry/app/views/onboarding/configure/index.jsx [hunk 1]
```diff
@@ -89,7 +88,6 @@ const Configure = createReactClass({
   },
 
   submit() {
-    HookStore.get('analytics:onboarding-complete').forEach(cb => cb());
```
Comment (denamwangi) [inline src/sentry/static/sentry/app/views/onboarding/configure/index.jsx]: is marketing capturing this a different way?

**Review by denamwangi — APPROVED** [review]

File: src/sentry/static/sentry/app/views/onboarding/configure/index.jsx (hunk 1 above)
Comment (adhiraj) [reply src/sentry/static/sentry/app/views/onboarding/configure/index.jsx]: don't think anyone is using this

**Comment by benvinegar** [comment]
Seems wrong to have Amplitude-specific code in this repository?

**Comment by adhiraj** [comment]
The actual code is in getsentry, amplitude is just the name of the hook. We could call it something generic like 'external-logging-hook' -- would that be better?

**Comment by adhiraj** [comment]
(I was trying to going for explicitness/clarity)

**Pull request merged**